### ISSUE TITLE
Incomplete Browserslist support in build

### ISSUE THREAD

**Issue by chrissantamaria** [gatsbyjs/gatsby#29943]
## Description

Gatsby v3 now by default seems to set a Webpack `target` of `browserslist:` followed by the raw config (either inferred from a Browserslist config or a fallback Gatsby one) for non-HTML build steps. This works well for single-environment configs though it unfortunately breaks for [multi-environment ones](https://github.com/browserslist/browserslist#configuring-for-different-environments).

From what I can tell, this is due to Gatsby always using the `defaults` environment:

https://github.com/gatsbyjs/gatsby/blob/528a459c25685475f949692a21f12d4c83a878a9/packages/gatsby/src/utils/browserslist.ts#L19-L32

With a multi-environment config, defaults may not be defined, resulting in `getBrowsersList` returning `[]` which causes unexpected behavior in the Webpack config.

One possible way around this would be to use [`browserslist.loadConfig`](https://github.com/browserslist/browserslist/blob/c8dc18a681e50562f6c5a814e9bb7e864e4c7874/node.js#L212-L227) instead. This would offer the flexibility of choosing environments through [environment variables](https://github.com/browserslist/browserslist#environment-variables) while still maintaining the existing functionality of falling back to `defaults` (see [logic in `pickEnv`](https://github.com/browserslist/browserslist/blob/c8dc18a681e50562f6c5a814e9bb7e864e4c7874/node.js#L67-L82))

The only wrinkle I see is that `loadConfig` can return a string rather than an array when a config is set through the `BROWSERSLIST` environment variable - this case can be handled though.

This issue is also present in Gatsby v2 (logic in `getBrowsersList` is the same), though I haven't encountered a case where it's caused a fatal error.

### Steps to reproduce

I've created a [reproduction repo](https://github.com/chrissantamaria/gatsby-getbrowserslist-bug) with steps of how to produce this bug.

### Expected result

The `build` script should succeed and use the `production` Browserslist config (based on `process.env.NODE_ENV === 'production'`)

The `build:modern` script should succeed and use the `modern` Browserslist config (based on `process.env.BROWSERSLIST_ENV === 'modern'` )

All other Browserslist environment variables should be respected as well.

### Actual result

The build fails due to an invalid Webpack config.

### Environment

```
  System:
    OS: macOS 10.15.7
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 14.15.3 - ~/.volta/tools/image/node/14.15.3/bin/node
    Yarn: 1.22.10 - ~/.volta/tools/image/yarn/1.22.10/bin/yarn
    npm: 6.14.9 - ~/.volta/tools/image/node/14.15.3/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 88.0.4324.192
    Firefox: 78.0.1
    Safari: 14.0.2
  npmPackages:
    gatsby: ^3.0.0 => 3.0.0
```

**Comment by chrissantamaria** [comment]
@wardpeet thanks for the quick fix! I understand that the Webpack config bug in v3 is the most pressing, but is the core issue (`getBrowsersList` not properly handling multi-environment configs) planned to be addressed as well?

**Comment by chrissantamaria** [comment]
@wardpeet 👋  just wanted to bump this. I'm happy to open a PR with the suggested approach above if you think that's the best way forward. This (afaik) would allow for bringing back `target: 'browserslist'` for Webpack while also supporting multi-environment Browserslist configs.

**Comment by chrissantamaria** [comment]
@wardpeet @LekoArts sorry for the ping, just wanted to bump this issue another time. I realize it's a bit old but I believe the underlying issue is still present here. This has now become apparent by builds failing when calling `hasES6ModuleSupport` (which, like the previous Webpack target check, fails when `getBrowsersList` returns `[]`).

I've updated the [reproduction repo](https://github.com/chrissantamaria/gatsby-getbrowserslist-bug) to use Gatsby v4.9.1 for easier troubleshooting. I've also tested the `browserslist.loadConfig` fix described in my original post and can confirm it still works - appears to be a simple change which would fix the `hasES6ModuleSupport` issue and allow for returning to `browserslist` as a Webpack target.

Happy to clarify any details about my use case that might be helpful for troubleshooting. Thanks!
